<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="JP Sinnwell, DJ Schaid" />


<title>Penalized Variance Components</title>


<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Penalized Variance Components</h1>
<h4 class="author">JP Sinnwell, DJ Schaid</h4>


<div id="TOC">
<ul>
<li><a href="#overview-of-the-vcpen-package-for-penalized-variance-components">Overview of the <em>vcpen</em> package for Penalized Variance Components</a></li>
<li><a href="#preparing-to-run-vcpen">Preparing to run <em>vcpen</em></a><ul>
<li><a href="#sample-dataset">Sample dataset</a></li>
<li><a href="#make-kernel-matrices">Make kernel matrices</a></li>
</ul></li>
<li><a href="#penalized-estimation-of-vcs">Penalized estimation of VCs</a><ul>
<li><a href="#default-settings.">Default settings.</a></li>
<li><a href="#changing-penalty-fraction">Changing penalty fraction:</a></li>
</ul></li>
<li><a href="#demo-of-using-minque-outside-of-vcpen">Demo of using <code>minque()</code> outside of <code>vcpen()</code></a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<div id="overview-of-the-vcpen-package-for-penalized-variance-components" class="section level1">
<h1>Overview of the <em>vcpen</em> package for Penalized Variance Components</h1>
<p>A penalized likelihood model is used to estimate variance components with an elastic-net penalty function that applies both L1 and L2 penalties to the variance components, using the function <code>vcpen()</code>. Each variance component multiplies a kernel matrix, and we provide the function <code>kernel_linear()</code> to compute linear kernel matrices, but users are welcome to use their own functions to compute kernel matrices.</p>
<p>The function <code>vcpen()</code> allows the user to provide intitial starting values for the variance components. If no initial values are provided, the default is to use our funcion <code>minque()</code> to calculate initial values. For linear mixed models, MINQUE is the first iteration of restricted maximum likeihood estimation (REML), and iterative updates of MINQUE converge to REML estimation.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">require</span>(vcpen)</a></code></pre></div>
</div>
<div id="preparing-to-run-vcpen" class="section level1">
<h1>Preparing to run <em>vcpen</em></h1>
<div id="sample-dataset" class="section level2">
<h2>Sample dataset</h2>
<p>Below provides snapshots of an example dataset. The response is the outcome variable, covmat is a matrix of adjusting covariates, and dose is a matrix of the dose of a minor allele for SNPs (dose values of 0, 1, 2). The doseinfo illustrates how the SNPs (columns of dose) map into groups, for creating kernel matrices for each group. A kernel matrix for n subjects is an nxn matrix that measures similarity of the dose values for each pair of subjects.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">data</span>(vcexample)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">ls</span>()</a></code></pre></div>
<pre><code>[1] &quot;covmat&quot;   &quot;dose&quot;     &quot;doseinfo&quot; &quot;response&quot;</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">head</span>(dose)</a></code></pre></div>
<pre><code>  snp1 snp2 snp3 snp4 snp5 snp6 snp7 snp8 snp9 snp10 snp11 snp12 snp13 snp14
1    0    0    0    0    1    1    1    0    0     1     1     0     0     1
2    1    0    0    0    0    1    0    0    0     0     2     0     0     1
3    0    0    0    0    1    0    0    0    0     1     0     0     0     0
4    1    0    0    0    1    0    0    1    1     0     0     0     0     0
5    0    0    0    0    0    1    0    0    0     1     0     0     1     0
6    0    0    0    0    1    0    0    0    0     0     1     0     0     0
  snp15 snp16 snp17 snp18 snp19 snp20 snp21 snp22 snp23 snp24 snp25 snp26 snp27
1     1     0     1     1     0     1     2     0     0     0     0     1     0
2     2     1     0     1     0     1     0     0     0     0     0     0     0
3     0     0     0     0     0     0     0     1     1     1     0     1     0
4     0     1     0     0     1     1     0     0     0     0     0     0     0
5     0     0     0     0     0     0     1     0     0     1     0     1     0
6     1     0     0     0     0     0     0     1     0     0     1     0     0
  snp28 snp29 snp30 snp31 snp32 snp33 snp34 snp35 snp36 snp37 snp38 snp39 snp40
1     0     0     0     0     0     1     0     0     0     0     0     0     1
2     0     0     1     0     0     0     1     0     0     0     0     1     0
3     0     1     0     0     0     0     0     0     0     0     1     1     0
4     1     1     0     0     0     0     0     0     0     0     0     1     0
5     0     0     0     0     0     1     1     0     0     0     0     0     1
6     0     0     0     0     0     0     0     1     0     0     0     0     0
  snp41 snp42 snp43 snp44 snp45 snp46 snp47 snp48 snp49 snp50 snp51 snp52 snp53
1     1     0     1     0     1     1     0     1     1     0     0     0     0
2     0     1     0     1     0     1     0     1     0     0     0     0     0
3     0     1     0     0     0     1     1     0     0     0     0     0     0
4     0     0     0     0     0     1     0     0     0     0     0     1     0
5     0     1     0     1     0     0     0     1     0     1     1     1     0
6     0     1     0     0     1     0     1     0     0     0     0     2     0
  snp54 snp55 snp56 snp57 snp58 snp59 snp60 snp61 snp62 snp63 snp64 snp65 snp66
1     0     0     1     0     0     0     0     0     0     0     0     0     0
2     0     0     0     0     0     1     0     0     1     0     0     1     0
3     0     0     1     0     0     0     0     0     1     0     0     0     0
4     0     0     0     0     1     0     1     0     1     0     0     1     0
5     0     0     0     0     0     0     0     0     1     0     0     0     0
6     1     0     1     0     1     0     0     0     0     2     0     1     0
  snp67 snp68 snp69 snp70
1     0     0     0     0
2     0     1     0     0
3     1     0     0     0
4     0     0     0     0
5     0     0     0     0
6     0     0     0     0</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">head</span>(doseinfo)</a></code></pre></div>
<pre><code>  index vcname
1     1    vc1
2     1    vc1
3     1    vc1
4     1    vc1
5     1    vc1
6     1    vc1</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">response[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</a></code></pre></div>
<pre><code> [1]  0.54166685  0.07233516 -1.03718603 -1.16407584  1.17964672  0.14994286
 [7]  1.82625006 -2.83793412 -0.18631904  1.53431587</code></pre>
</div>
<div id="make-kernel-matrices" class="section level2">
<h2>Make kernel matrices</h2>
<p>The example below illustrates how to loop over groups (indicated by doseinfo) to create linear kernel matrices for each group. Note that the number of variance components is the number of groups plus 1, because the last group is for the residual variance component, which will have a kernel matrix that is the identity matrix.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">nvc &lt;-<span class="st"> </span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(doseinfo[, <span class="dv">2</span>]))</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">id &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(dose)</a>
<a class="sourceLine" id="cb10-3" data-line-number="3"></a>
<a class="sourceLine" id="cb10-4" data-line-number="4">## vcs for genetic kernel matrices</a>
<a class="sourceLine" id="cb10-5" data-line-number="5">Kerns &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="st">&quot;list&quot;</span>, <span class="dt">length =</span> nvc)</a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(nvc <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)) {</a>
<a class="sourceLine" id="cb10-7" data-line-number="7">    ## below uses kernel_linear, but users can replace this with their choice</a>
<a class="sourceLine" id="cb10-8" data-line-number="8">    ## of function to create other types of kernel matrices.</a>
<a class="sourceLine" id="cb10-9" data-line-number="9">    Kerns[[i]] &lt;-<span class="st"> </span><span class="kw">kernel_linear</span>(dose[, <span class="kw">grep</span>(i, doseinfo[, <span class="dv">2</span>])])</a>
<a class="sourceLine" id="cb10-10" data-line-number="10">    <span class="kw">rownames</span>(Kerns[[i]]) &lt;-<span class="st"> </span>id</a>
<a class="sourceLine" id="cb10-11" data-line-number="11">    <span class="kw">colnames</span>(Kerns[[i]]) &lt;-<span class="st"> </span>id</a>
<a class="sourceLine" id="cb10-12" data-line-number="12">}</a>
<a class="sourceLine" id="cb10-13" data-line-number="13">## vc for residual variance requires identity matrix</a>
<a class="sourceLine" id="cb10-14" data-line-number="14">Kerns[[nvc]] &lt;-<span class="st"> </span><span class="kw">diag</span>(<span class="kw">nrow</span>(dose))</a>
<a class="sourceLine" id="cb10-15" data-line-number="15"><span class="kw">rownames</span>(Kerns[[nvc]]) &lt;-<span class="st"> </span>id</a>
<a class="sourceLine" id="cb10-16" data-line-number="16"><span class="kw">colnames</span>(Kerns[[nvc]]) &lt;-<span class="st"> </span>id</a></code></pre></div>
</div>
</div>
<div id="penalized-estimation-of-vcs" class="section level1">
<h1>Penalized estimation of VCs</h1>
<div id="default-settings." class="section level2">
<h2>Default settings.</h2>
<p>Run with default settings, which uses <code>minque()</code> to estimate initial values for variance components and default <code>frac1=0.8</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">fit &lt;-<span class="st"> </span><span class="kw">vcpen</span>(response, covmat, Kerns)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw">summary</span>(fit)</a></code></pre></div>
<pre><code>vcpen object
  N-subjects = 100
  N-VC = 5

  Model fits over lambda penalty grid:

   lambda iter   logl loglpen   bic min_bic
1    0.10   49 -155.2   160.5 333.4   FALSE
2    0.09   30 -154.7   159.9 332.4   FALSE
3    0.08   24 -154.3   159.3 331.5   FALSE
4    0.07   20 -153.9   158.7 330.8   FALSE
5    0.06   18 -153.6   158.0 330.2   FALSE
6    0.05   16 -153.3   157.2 329.7   FALSE
7    0.04   15 -153.1   156.4 329.3   FALSE
8    0.03   14 -152.9   155.6 328.9   FALSE
9    0.02   13 -152.8   154.7 328.6   FALSE
10   0.01   13 -152.7   153.7 328.5    TRUE
11   0.00  110 -152.7   152.7 333.0   FALSE

  VC estimates by lambda penalties:

   lambda     vc1    vc2       vc3       vc4    vc5
1    0.10 0.06318 0.1301 0.000e+00 0.000e+00 1.0854
2    0.09 0.07712 0.1535 0.000e+00 2.435e-20 1.0523
3    0.08 0.09139 0.1779 0.000e+00 0.000e+00 1.0231
4    0.07 0.10522 0.2020 0.000e+00 0.000e+00 0.9996
5    0.06 0.11933 0.2269 1.448e-20 0.000e+00 0.9796
6    0.05 0.13405 0.2530 0.000e+00 0.000e+00 0.9622
7    0.04 0.14992 0.2813 0.000e+00 0.000e+00 0.9466
8    0.03 0.16730 0.3123 0.000e+00 0.000e+00 0.9326
9    0.02 0.18676 0.3472 0.000e+00 1.028e-21 0.9200
10   0.01 0.20925 0.3875 2.605e-22 0.000e+00 0.9082
11   0.00 0.23670 0.4372 2.420e-13 5.713e-03 0.8910

Estimates with min BIC:
beta:
   x0    x1 
0.628 1.097 
VC estimates:
      vc1    vc2       vc3 vc4    vc5
10 0.2092 0.3875 2.605e-22   0 0.9082</code></pre>
</div>
<div id="changing-penalty-fraction" class="section level2">
<h2>Changing penalty fraction:</h2>
<p>Perform the same run as above, but with lower penalty fraction.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">fit.frac1 &lt;-<span class="st"> </span><span class="kw">vcpen</span>(response, covmat, Kerns, <span class="dt">frac1 =</span> <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="kw">summary</span>(fit.frac1)</a></code></pre></div>
<pre><code>vcpen object
  N-subjects = 100
  N-VC = 5

  Model fits over lambda penalty grid:

   lambda iter   logl loglpen   bic min_bic
1    0.10   25 -154.2   157.4 331.5   FALSE
2    0.09   14 -154.1   157.1 331.1   FALSE
3    0.08   15 -153.8   156.7 330.7   FALSE
4    0.07   14 -153.6   156.3 330.3   FALSE
5    0.06   14 -153.4   155.9 329.9   FALSE
6    0.05   13 -153.3   155.5 329.5   FALSE
7    0.04   13 -153.1   155.0 329.2   FALSE
8    0.03   13 -152.9   154.5 328.9   FALSE
9    0.02   12 -152.8   154.0 328.7   FALSE
10   0.01   12 -152.7   153.4 328.5    TRUE
11   0.00  110 -152.7   152.7 333.0   FALSE

  VC estimates by lambda penalties:

   lambda     vc1    vc2      vc3       vc4    vc5
1    0.10 0.09851 0.1707 0.00e+00 0.000e+00 1.0213
2    0.09 0.10547 0.1832 0.00e+00 0.000e+00 1.0101
3    0.08 0.11416 0.1988 0.00e+00 0.000e+00 0.9959
4    0.07 0.12345 0.2158 0.00e+00 0.000e+00 0.9825
5    0.06 0.13371 0.2347 0.00e+00 0.000e+00 0.9694
6    0.05 0.14496 0.2556 0.00e+00 0.000e+00 0.9569
7    0.04 0.15772 0.2797 0.00e+00 0.000e+00 0.9445
8    0.03 0.17232 0.3076 3.16e-23 0.000e+00 0.9324
9    0.02 0.18924 0.3406 0.00e+00 1.706e-23 0.9208
10   0.01 0.20977 0.3815 0.00e+00 0.000e+00 0.9091
11   0.00 0.23670 0.4372 2.42e-13 5.713e-03 0.8910

Estimates with min BIC:
beta:
   x0    x1 
0.628 1.097 
VC estimates:
      vc1    vc2 vc3 vc4    vc5
10 0.2098 0.3815   0   0 0.9091</code></pre>
</div>
</div>
<div id="demo-of-using-minque-outside-of-vcpen" class="section level1">
<h1>Demo of using <code>minque()</code> outside of <code>vcpen()</code></h1>
<p>This demonstrates how users can use <code>minque()</code> as a general approach to approximate REML variance components. Increasing <code>n.iter</code> will cause the resulting variance components to be closer to the fully interative REML estimates.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">vcinit &lt;-<span class="st"> </span><span class="kw">minque</span>(response, covmat, Kerns, <span class="dt">n.iter =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw">names</span>(vcinit)</a></code></pre></div>
<pre><code>[1] &quot;vc&quot;        &quot;beta&quot;      &quot;residuals&quot;</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">vcinit<span class="op">$</span>beta</a></code></pre></div>
<pre><code>          [,1]
[1,] 0.6279622
[2,] 1.0961475</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">vcinit<span class="op">$</span>vc</a></code></pre></div>
<pre><code>          [,1]
[1,] 0.2279593
[2,] 0.4254118
[3,] 0.0000000
[4,] 0.0000000
[5,] 1.0161211</code></pre>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<p>Schaid DJ, Sinnwell JP, Larson NB, Chen J (2020). Penalized Variance Components for Association of Multiple Genes with Traits. Genet Epidemiol, To Appear.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
